### ISSUE TITLE
Svelte: support self-referencing component

### ISSUE THREAD

**Issue by samijaber** [BuilderIO/mitosis#291]
For a component to recursively reference itself, it needs to call `<svelte:self>` instead of `<component-name>`: https://svelte.dev/tutorial/svelte-self

We need to bake this logic into the Svelte generator, specifically the `blockToSvelte` function: 

https://github.com/BuilderIO/mitosis/blob/fbedeeedd016df678bbc92be3143c272a3d5127c/packages/core/src/generators/svelte.ts#L54

We should be able to compare `json.name` of the _block_ against the `name` of the original root Mitosis JSON, and if they're equal, replace the value with `'svelte:self'`

**Comment by samijaber** [comment]
Example: https://mitosis.builder.io/?outputTab=svelte&code=import+%7B+useState%2C+Show+%7D+from+%22%40builder.io%2Fmitosis%22%3B%0A%0Aexport+default+function+MyComponent%28props%29+%7B%0A++return+%28%0A++++%3Cdiv%3E%0A++++++%7Bprops.name%7D%0A++++++%3CShow+when%3D%7Bprops.name+%3D%3D%3D+%27Sami%27%7D%3E%0A++++++++%3CMyComponent+name%3D%7B%27Jaber%27%7D+%2F%3E%0A++++++%3C%2FShow%3E%0A++++%3C%2Fdiv%3E%0A++%29%3B%0A%7D

Current output:

```svelte
<script>
  export let name;

</script>

<div>
  {name}

  {#if name === 'Sami'}
    <MyComponent name={'Jaber'} />
  {/if}
</div>
```

expected output:

```svelte
<script>
  export let name;

</script>

<div>
  {name}

  {#if name === 'Sami'}
    <svelte:self name={'Jaber'} />
  {/if}
</div>

```

**Comment by sanyamkamat** [comment]
Fixed by #293
